fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/LoanDelinquencyDomainServiceImpl.java

```diff
@@ -141,7 +141,12 @@ public CollectionData getOverdueCollectionData(final Loan loan) {
         collectionData.setDelinquentAmount(outstandingAmount);
 
         if (overdueDays > 0) {
-            collectionData.setDelinquentDays(overdueDays - graceDays);
+            Long delinquentDays = 0L;
```

Comment:
So what's the point of this variable? Why aren't you doing is a simpler way?

```
collectionData.setDelinquentDays(0);
Long delinquentDays = overdueDays - graceDays;
if (delinquentDays > 0) {
      collectionData.setDelinquentDays(delinquentDays);
}
```
```

Reply:
I need to evaluate if overdue days has value, It can be null